### ISSUE TITLE
Completely broken on Firefox, likely due to recent YT UI update

### ISSUE THREAD

**Issue by BobBobBobBobBobBobBobBobBob** [YePpHa/YouTubeCenter#2408]
Currently does not work at all for me. I'm running the dev version via greasemonkey on the current build of Firefox, and every time I go to watch a video with YTC running, all I get is a blank page. Moreover, the settings menu is totally unavailable.

I don't have any other browsers with it installed, although spoofing Firefox's browser profile to any number of other web browsers does nothing to mitigate the problem.

**Comment by p1nkyy** [comment]
+1 for me. FF 53, GM 3.10, YTC 548

**Comment by p1nkyy** [comment]
No, it's not. You made me install TM, gahh.. Video auto-played when I have disabled that option, and annotation is on when it shouldn't be.

**Comment by omperus** [comment]
Same here, video autoplays when i have the option disabled.
Sometimes it stops 2 seconds in (automatically) and i have to refresh the page because the play button doesn't work anymore.

**Comment by joaocpfb** [comment]
On Apr 21, 2017 2:06 AM, "BobBobBobBobBobBobBobBobBob" <
notifications@github.com> wrote:
>
> Currently does not work at all for me. I'm running the dev version via
greasemonkey on the current build of Firefox, and every time I go to watch
a video with YTC running, all I get is a blank page. Moreover, the settings
menu is totally unavailable.
>
> I don't have any other browsers with it installed, although spoofing
Firefox's browser profile to any number of other web browsers does nothing
to mitigate the problem.

**Comment by JohanPirlouit** [comment]
> greasemonkey on the current build of Firefox, and every time I go to watch a video with YTC running, 
all I get is a blank page. Moreover, the settings menu is totally unavailable.

Hi,

I had the same with FF 51 + Greasemonkey... Btw, at this time I could revert back to the previous YT UI through a link that was somewere I don't remerber where it was on the new UI :-/ ... Since, I still NOT use the new YT UI and YTC still work fine, except that YT/Google seem to have locked some protected content (see #2396).

Regards

**Comment by BobBobBobBobBobBobBobBobBob** [comment]
@sdsucks 

>dev 548 working fine in FF 53 with tampermonkey 4.3.5393

Didn't solve anything for me. I was able to access the YTC options menu through Tampermonkey, but it was basically broken. Oddly, however, as autoplay was enabled by default on running it via Tampermonkey, videos seem to play in the background even though the page I'm given is still completely blank.

I've also disabled any extensions/scripts that could possibly interfere to no avail.

**Comment by p1nkyy** [comment]
1. How to quote in github? Please teach! 😃 
2. @joaocpfb I also had the same problem, video appear in only a black line but it's playing, I reloaded the page everytime I encountered this and then the normal video window will appear.

**Comment by BobBobBobBobBobBobBobBobBob** [comment]
It *appears* to be functional now.

@p1nkyy 

Just start the line with [>] (minus brackets)

**Comment by p1nkyy** [comment]
Not in me, sometimes it works, sometimes not.

**Comment by omperus** [comment]
It's still completely broken for me.

**Comment by p1nkyy** [comment]
Seriously considering to e-mail the dev right now 😖

**Comment by Parsnip** [comment]
Yeah, for me annotations and video autoplay are the most annoying problems right now on the normal youtube interface. Sounds like the new material design youtube that they are testing completely breaks it though.

**Comment by p1nkyy** [comment]
Someone in the chrome/chromium thread found a solution,
1. to change all the con.log words in the userscripts into console.log
2. Change something into bootstrap.
The 2nd step, I don't know the equivalent in Firefox, maybe someone can help?
Check: https://github.com/YePpHa/YouTubeCenter/issues/2407

**Comment by omperus** [comment]
@p1nkyy @Parsnip @BobBobBobBobBobBobBobBobBob @JohanPirlouit @joaocpfb @sdsucks 

Just tried that solution and it worked! 😀 **(Windows 10 64-bit Firefox 53.0 64-bit)**

Go to C:\Users\YOUR USERNAME\AppData\Roaming\Mozilla\Firefox\Profiles\YOUR USERNAME\gm_scripts\YouTube_Center_Developer_Build

There edit YouTubeCenter.user.js (i used notepad++) press CTRL+F, go to the Replace tab, in find input "con.log" and in replace input "console.log", hit "Replace All" (it should replace about 144 entries) then you save and restart Firefox!

I hope this helps.

**Comment by p1nkyy** [comment]
Doesn't really work for me. Maybe someone else can confirm? I still think it needs the 2nd step...

**Comment by Parsnip** [comment]
I'm not quite sure that chrome issue is the same issue at all since it apparently works in older chromium.

And in this thread we have at least 2 different issues (completely broken with just blank youtube page versus mostly working with broken autoplay and annotations).

**Comment by p1nkyy** [comment]
Maybe it's time to summon the developers: @YePpHa @aggroskater , please help us that's desperately want this awesome script to work again!

**Comment by omperus** [comment]
@Parsnip I get the blank page too, just not too often.

**Comment by sdsucks** [comment]
weird because dev548 + FF54b2 + GM3.11 is working just fine for me

**Comment by g1zstar** [comment]
Used same as above except beta 3 instead of 2 for FF. I don't think it's completely broken for me but something I've noticed is if I have Tab Auto Pause on and I allow it to do that, it'll permanently pause the video. I have to refresh the window to be able to play the video.

**Comment by CollinChaffin** [comment]
@omperus it is probably only broken again because, if you all perform this local on-disk edit, you should either also manually increment the version of the script internally (BAD IDEA because then you won't get the real update when it comes), or better idea - DISABLE ALL AUTO-UPDATING OF SCRIPTS in GM/TM, or your local edits will simply be silently overwritten on the next update check.

**Comment by jonnyrobbie** [comment]
I have to confirm the issue. YTC548, FF nightly 55.0a1 (2017-05-01) (64-bit), linux.

**Comment by zmwangx** [comment]
The blank page issue is due to a UI redesign, which I've been seeing in increased frequency for the past few months. And now YouTube have made it official: https://youtube.googleblog.com/2017/05/a-sneak-peek-at-youtubes-new-look-and.html.

**Comment by rayman89** [comment]
**Replacing on the script "con.log" for "console.log" on firefox fixes a lot of stuff.**

**Comment by Yaro96** [comment]
For those having problems with stopping auto-play, I managed to fix this by commenting the 2037th line of the script
`//originalPlayFunc.apply(this, arguments);`
I'm not entirely sure how this works, but until an official patch from @YePpHa, it does the job done for me.

**Comment by Parsnip** [comment]
No such luck here. I seem to be getting the same behavior whether that's commented out or not. Shame.
